src/Sparrow.Server/Allocators.PointerTypes.cs

```diff
@@ -0,0 +1,527 @@
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Sparrow.Server
+{
+    public interface IPointerType { }
+    public interface IPointerType<T> : IPointerType where T : struct { }
+
+    [StructLayout(LayoutKind.Sequential)]
+#if !VALIDATE
+    // We do this to have the ability to perform extensive validation for pointers.
+    readonly
+#endif
+    public unsafe struct Pointer : IPointerType, Meta.IDescribe
+        {
```

Comment:
Need reformating

Reply:
Yeah, still every time you change something it gets screwed again... VS support for #define statements is quite crude for C#.